equipment-service/src/main/java/com/telecominfraproject/wlan/equipment/controller/EquipmentController.java

```diff
@@ -354,6 +361,90 @@ public Equipment updateChannels(@RequestBody EquipmentChannelsUpdateRequest requ
         return ret;
     }
     
+    /**
+     * Updates Equipment CellSizeAttributes
+     *
+     * @param EquipmentCellSizeAttributesUpdateRequest
+     * @return updated Equipment object
+     * @throws RuntimeException if Equipment record does not exist or if it was modified concurrently
+     */
+    @RequestMapping(value = "/cellSize", method=RequestMethod.PUT)
+    public Equipment updateCellSizeAttributes(@RequestBody EquipmentCellSizeAttributesUpdateRequest request) {
+        
+        LOG.debug("updateCellSizeAttributes {}", request);
+        if (request == null) {
+            return null;
+        }
+        Equipment existingEquipment = get(request.getEquipmentId());
+        
+        if (existingEquipment.getDetails() == null) {
+            LOG.info("updateCellSizeAttributes: no details on equipment");
+            return null;
+        }
+        Equipment equipmentCopy = existingEquipment.clone();
+        
+        Map<RadioType, CellSizeAttributes> cellSizeAttributesMap = request.getCellSizeAttributesMap();
+        Map<RadioType, Boolean> autoCellSizeSelections = request.getAutoCellSizeSelections();
+        
+        if (CollectionUtils.isEmpty(autoCellSizeSelections) ||
+                CollectionUtils.isEmpty(cellSizeAttributesMap)) {
+            LOG.info("updateCellSizeAttributes no update");
+            return equipmentCopy;
+        }
+        
+        ApElementConfiguration apElementConfiguration = (ApElementConfiguration) equipmentCopy.getDetails();
+        
+        for (RadioType radioType : RadioType.validValues()) {
+            ElementRadioConfiguration elementRadioConfig = apElementConfiguration.getElementRadioConfiguration(radioType);
+            RadioConfiguration radioConfig = apElementConfiguration.getAdvancedRadioMap().get(radioType);
+            if (elementRadioConfig == null || radioConfig == null) {
+                continue;
+            }
+            CellSizeAttributes cellSizeAttributes = cellSizeAttributesMap.get(radioType);
+            if (cellSizeAttributes == null) {
+                LOG.info("updateCellSizeAttributes cellSizeAttributes is null for radioType {}", radioType);
+                continue;
+            }
+            if (cellSizeAttributesMap.get(radioType) != null && autoCellSizeSelections.get(radioType) != null) {
```

Comment:
Small thing but since we just checked cellSizeAttributes == null then I don't think we should have to check 'cellSizeAttributesMap.get(radioType) != null'